go/libraries/doltcore/sqle/tables.go

```diff
@@ -1971,76 +1984,80 @@ func (t *AlterableDoltTable) AddForeignKey(ctx *sql.Context, sqlFk sql.ForeignKe
 			refColTags[i] = refCol.Tag
 		}
 
+		var tableIndexName, refTableIndexName string
 		tableIndex, ok, err := findIndexWithPrefix(t.sch, sqlFk.Columns)
 		if err != nil {
 			return err
 		}
+		// check if foreign key is prefix of primary key
 		if !ok {
-			// The engine matched on a primary key, and Dolt does not yet support using the primary key within the
-			// schema.Index interface (which is used internally to represent indexes across the codebase). In the
-			// meantime, we must generate a duplicate key over the primary key.
-			//TODO: use the primary key as-is
-			idxReturn, err := creation.CreateIndex(ctx, tbl, "", sqlFk.Columns, false, false, "", editor.Options{
-				ForeignKeyChecksDisabled: true,
-				Deaf:                     t.opts.Deaf,
-				Tempdir:                  t.opts.Tempdir,
-			})
-			if err != nil {
-				return err
-			}
-			tableIndex = idxReturn.NewIndex
-			tbl = idxReturn.NewTable
-			root, err = root.PutTable(ctx, t.tableName, idxReturn.NewTable)
-			if sqlFk.IsSelfReferential() {
-				refTbl = idxReturn.NewTable
+			if !isPkPrefix(sqlFk.Columns, t.sch.GetPKCols().GetColumns()) {
```

Comment:
This seems incorrect. This should _only_ get to this state when the engine matched a primary key, as that's the only time that the engine found an index but Dolt does not have the index. So I think this block should assume that the primary key IS the target index. With this check, it seems like there's an opportunity for some other additional index to be created, which would be a bug. We should also keep the comment stating as such (of course updated).

General guidance regarding comments when you're editing code, if you see a comment that no longer applies (and isn't just a "TODO"), it's better to update the comment than to remove it. Generally they're placed in sections that are worth documenting, so removing the comment may make it much harder to reason about some section of code in the future. This is a good example, as we know that this block should be "primary key only" _because_ of the old comment.

Reply:
I think the `!ok` from `findIndexWithPrefix` is just when there isn't a `SECONDARY INDEX` that matches; it is still possible that the columns used in the `FOREIGN KEY` are not a prefix of the `PRIMARY KEY`, which is why I added the `isPkPrefix` check, and a new `SECONDARY INDEX` should be made.

I can update the comment to reflect this.

Reply:
Nevermind, MySQL only creates indexes for the table, and not the referenced table.